### ISSUE TITLE
Read generator from the CMakeCache.txt

### ISSUE THREAD

**Issue by PinkySan** [microsoft/vscode-cmake-tools#1400]
### Brief Issue Summary
I am configuring my cmake-project with conan. I am having some special profiles for different compiler.
When opening the folder and trying to use the cmake-tools, i always have to select a kit. These seem sometimes to differ. Especially when i am using platform and toolsets. 
The kit/generator should be automatically selected by the already generated cmake-cache. 

### Expected:

1. Having a possibility to first check in the cmake.buildDirectory, if there is already a CMakeCache.txt
2. Setting the kit, generator, toolset, platform and cmake-executable regarding the cache (using different cmake-versions in different projects)

### Platform and Versions

- **Operating System**: Windows
- **CMake Version**: 3.10 or newer
- **VSCode Version**: 1.47.3
- **CMake Tools Extension Version**: v1.4.1
- **Compiler/Toolchain**: differ

### Other Notes/Information

also having problems with intellisense and conan
Neverthelesse: Good tool by the way 😉

**Comment by bobbrow** [comment]
Good suggestion. We'll put it on our backlog.

**Comment by codewing** [comment]
I'm also using Conan with a bundled Android NDK that is not detected automatically. 
For me this would make this great extension even better.

**Comment by KrisThielemans** [comment]
obviously, it should also use existing configurations in the `CMakeCache.txt` (I'm mentioning just to be sure).

[We use a superbuild](https://github.com/SyneRBI/SIRF-SuperBuild/) that is just too complicated to configure from Code. All our projects are added via `ExternalProject_Add`. It'd be nice to either just run the superbuild from Code, or open the `CMakeCache.txt` from one of the projects, and just work on that one.
